app/src/main/java/ch/epfl/skysync/viewmodel/FinishedFlightsViewModel.kt

```diff
@@ -115,6 +126,45 @@ class FinishedFlightsViewModel(val repository: Repository, val userId: String) :
             }
       }
 
+  private fun searchLocation(query: String) =
+      viewModelScope.launch(Dispatchers.IO) {
+        val client = OkHttpClient()
+        val url = "https://nominatim.openstreetmap.org/search?q=$query&format=json&limit=4"
+        val request = Request.Builder().url(url).build()
+        _searchResponse.value =
+            try {
+              val response = client.newCall(request).execute()
+              val responseBody = response.body()?.string()
+              responseBody
+            } catch (e: Exception) {
+              e.printStackTrace()
+              "error"
+            }
+      }
+
+  fun getSearchLocation(query: String, time: Int) =
+      viewModelScope.launch {
+        searchLocation(query).join()
+        val gson = Gson()
+        if (_searchResponse.value == null) {
```

Comment:
Does `_searchResponse` need to be a flow ? Couldn't you return the response from the `searchLocation` function, `_searchResults` is a flow and gets updated (also `_searchResponse` is private and never exposed)

Reply:
No, it is not needed. I have change that.